packages/agreement-process/test/testUpgradeAgreement.ts

```diff
@@ -0,0 +1,1339 @@
+        Buffer.from("large-document-file")
+      );
+      expect(await fileManager.listFiles(config.s3Bucket)).toContainEqual(
+        `${config.consumerDocumentsPath}/${path}/${name}`
+      );
+    }
+
+    it("should succeed with valid Verified and Declared attributes", async () => {
+      const authData = {
+        ...getRandomAuthData(),
+        userRoles: [userRoles.ADMIN_ROLE],
+      };
+      const tenantId = authData.organizationId.toString();
+
+      const validVerifiedTenantAttribute = {
+        ...getMockVerifiedTenantAttribute(unsafeBrandId<AttributeId>(tenantId)),
+        verifiedBy: [
+          {
+            id: unsafeBrandId<TenantId>(tenantId),
+            verificationDate: new Date(TEST_EXECUTION_DATE.getFullYear() - 1),
+            expirationDate: new Date(TEST_EXECUTION_DATE.getFullYear() + 1),
+            extensionDate: undefined,
+          },
+        ],
+      };
+      const validVerifiedEserviceAttribute = getMockEServiceAttribute(
+        validVerifiedTenantAttribute.id
+      );
+      const validDeclaredTenantAttribute = {
+        ...getMockDeclaredTenantAttribute(unsafeBrandId<AttributeId>(tenantId)),
+        revocationTimestamp: undefined,
+      };
+      const validDeclaredEserviceAttribute = getMockEServiceAttribute(
+        validDeclaredTenantAttribute.id
+      );
+
+      const validCertifiedTenantAttribute = getMockCertifiedTenantAttribute(
+        unsafeBrandId<AttributeId>(tenantId)
+      );
+      const validCertifiedEserviceAttribute = getMockEServiceAttribute(
+        validCertifiedTenantAttribute.id
+      );
+
+      const descriptorId = generateId<DescriptorId>();
+      const deprecatedDescriptor = {
+        ...getMockDescriptorPublished(
+          descriptorId,
+          [[validCertifiedEserviceAttribute]],
+          [[validDeclaredEserviceAttribute]],
+          [[validVerifiedEserviceAttribute]]
+        ),
+        path: "/deprecatedDescriptor/doc",
+        state: descriptorState.deprecated,
+        version: "1",
+      };
+
+      const publishedDescriptor: Descriptor = {
+        ...getMockDescriptorPublished(
+          descriptorId,
+          [[validCertifiedEserviceAttribute]],
+          [[validDeclaredEserviceAttribute]]
+        ),
+        version: "2",
+      };
+
+      const tenant = getMockTenant(unsafeBrandId<TenantId>(tenantId), [
+        validCertifiedTenantAttribute,
+        validDeclaredTenantAttribute,
+        validVerifiedTenantAttribute,
+      ]);
+      const agreementToBeUpgraded: Agreement = {
+        ...getMockAgreement(
+          generateId<EServiceId>(),
+          unsafeBrandId<TenantId>(tenantId),
+          randomArrayItem(agreementUpgradableStates)
+        ),
+        descriptorId,
+        producerId: unsafeBrandId<TenantId>(tenantId),
+        stamps: {},
+        createdAt: TEST_EXECUTION_DATE,
+      };
+
+      const eservice = getMockEService(
+        agreementToBeUpgraded.eserviceId,
+        generateId<TenantId>(),
```

Comment:
I know it's theoretically possible to have this id different from the agreement `producerId`, but actually when agreements are created the `producerId` is always set to `eservice.producerId`, eservice and agreement producer are the same. So I would:
* set here the same id used for agreement producerId
* in order to have eservice.producerId and consumer different, so that you can test also certified attributes, I would make sure the tenant used for authdata and `consumerId` is different from this one used for producerId (create another tenant if needed)